packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -557,7 +657,11 @@ function computeRenderContext(inputs: RenderContextInputs, scrollPosition: Scrol
     // Clamp the value because the search may return an index out of bounds.
     // In the last index, this is not needed because Array.slice doesn't include it.
     const firstRowIndex = Math.min(
-      getNearestIndexToRender(inputs, top),
+      getNearestIndexToRender(inputs, top, {
+        atStart: true,
+        lastPosition:
+          inputs.rowsMeta.positions[inputs.rowsMeta.positions.length - 1] + inputs.lastRowHeight,
```

Comment:
Should we avoid the object wrapper here for performance reasons?
```diff
- options: SearchOptions | undefined = undefined,
+ lastPosition: number | undefined, // Or call it `startPosition`?
```

Reply:
This code is called once per render context update so I'm less worried because it's not multiplied by the number of rows/cells. The param name is ok as `lastPosition` considering what we do with it, but I'll check if I can simplify this logic.

Reply:
Feel free to discard then, I assumed it's being called more often 👍🏻